include/spdlog/logger.h

```diff
@@ -28,18 +28,32 @@
 #include <vector>
 
 #ifndef SPDLOG_NO_EXCEPTIONS
-#    define SPDLOG_LOGGER_CATCH()                                                                                                          \
+#    define SPDLOG_LOGGER_CATCH(location)                                                                                                  \
         catch (const std::exception &ex)                                                                                                   \
         {                                                                                                                                  \
-            err_handler_(ex.what());                                                                                                       \
+            if(location.filename)                                                                                                          \
+            {                                                                                                                              \
+                try                                                                                                                        \
+                {                                                                                                                          \
```

Comment:
I dont think an extra try catch needed here. why would it fail? the lib is in total control of the source loc param

Reply:
I just figured as fmt::format can throw (e.g. in case of OOM), it would be the safest bet to catch.
I'll remove it.

Reply:
Thanks, please remove it and I will merge.